app/models/plugins/thermometer.rb

```diff
@@ -28,4 +29,57 @@ def liquid_data
   def name
     self.class.name.demodulize
   end
+
+  def update_goal
+    self.goal = self.determine_next_goal
+    self.save
+  end
+
+  protected
```

Comment:
From where I stand, this should be `private`. Can you show me where you're making use of its `protected` status?